src/Routes.tsx

```diff
@@ -71,58 +67,22 @@ const routesOverhaul: Path[] = [
     }
 ];
 
-type InsightsRouteProps = Omit<RouteProps, 'component'> & Pick<Path, 'component'>;
-
-const InsightsRoute: React.FunctionComponent<InsightsRouteProps> = (props: InsightsRouteProps) => {
-    const { component, ...restProps } = props;
-    return (
-        <ErrorPage>
-            <Route { ...restProps }>
-                <CheckReadPermissions>
-                    <props.component />
-                </CheckReadPermissions>
-            </Route>
-        </ErrorPage>
-    );
-};
-
 export const Routes: React.FunctionComponent = () => {
-    const chrome = useChrome();
-    const history = useHistory();
     const notificationsOverhaul = useFlag('platform.notifications.overhaul');
 
     const pathRoutes = React.useMemo(() => notificationsOverhaul ? routesOverhaul : legacyRoutes, [ notificationsOverhaul ]);
 
-    React.useEffect(() => {
-        const on = chrome.on;
-        if (on) {
-            return on('APP_NAVIGATION', event => {
-                const pathname = event.domEvent.href;
-                const relative = pathname.substr(pathname.indexOf('/notifications'));
-                pathRoutes.some(({ path }) => {
-                    if (matchPath(relative, {
-                        path,
-                        exact: true
-                    })) {
-                        history.replace(relative);
-                        return true;
-                    }}
-                );
-            });
-        }
-    }, [ chrome.on, history, pathRoutes ]);
-
     return (
-        <Switch>
+        <DomRoutes>
             { pathRoutes.map(pathRoute => (
-                <InsightsRoute
-                    key={ pathRoute.path }
-                    component={ pathRoute.component }
-                    path={ pathRoute.path }
-                    exact={ true }
-                />
+                <Route key={ pathRoute.path } path={ pathRoute.path } element={ <CheckReadPermissions>
+                    <pathRoute.component />
+                </CheckReadPermissions> } />
             ))}
-            {!notificationsOverhaul && <RedirectToDefaultBundle />}
-        </Switch>
+            {!notificationsOverhaul && <Route
+                path="*"
```

Comment:
I am in favor of actually using 404 pages instead of unexpected redirects.

Reply:
I am as well, this is just backwards compatible layer. Notice the `notificationsOverhaul` once the notif overhaul will be GA this will be removed.